nautilus/include/nautilus/val_ptr.hpp

```diff
@@ -108,6 +108,12 @@ template <is_ptr ValuePtrType>
 class val<ValuePtrType> : public base_ptr_val<ValuePtrType> {
 public:
 	using base_ptr_val<ValuePtrType>::base_ptr_val;
+
+	// enable cast to type T
```

Comment:
Maybe specify how the cast looks like, e.g., 
```suggestion
	// enable cast to type T via static_cast<val<T>>(someVar);
```

Reply:
Would this also work?
```
val<uint32_t> val2 = 1234;
val<bool> val1 = static_cast<val<bool>>(val2);
```

Reply:
In this pr I only did it for pointer. But it should work. I will test it in another change